ui/src/app/edge/history/timeofusetariffdischarge/widget.component.ts

```diff
@@ -63,12 +60,21 @@ export class TimeOfUseTariffDischargeWidgetComponent extends AbstractHistoryWidg
         });
     }
 
-    protected getChannelAddresses(edge: Edge, config: EdgeConfig): Promise<ChannelAddress[]> {
+    protected override getChannelAddresses(edge: Edge, config: EdgeConfig): Promise<ChannelAddress[]> {
+        const result: ChannelAddress[] = [];
+        // Component was not initialized before, so had to initialize here.
+        this.service.getConfig().then(config => {
+            this.component = config.getComponent(this.componentId);
+        }).then(() => {
```

Comment:
chaining it 2 times is unecessary

Reply:
can it happen that no channels are sent with the promise at the end of the function?